homu/comments.py

```diff
@@ -18,6 +18,58 @@ def jsonify(self):
         return json.dumps(out, separators=(',', ':'))
 
 
+class Approved(Comment):
+    def __init__(self, bot=None, **args):
+        # Because homu needs to leave a comment for itself to kick off a build,
+        # we need to know the correct botname to use. However, we don't want to
+        # save that botname in our state JSON. So we need a custom constructor
+        # to grab the botname and delegate the rest of the keyword args to the
+        # Comment constructor.
+        super().__init__(**args)
+        self.bot = bot
+
+    params = ["sha", "approver"]
+
+    def render(self):
+        # The comment here is apparently required because Homu seems to use
+        # this note-to-self comment to kick off the build. Presumably because
+        # it contains the full commit hash?
```

Comment:
Btw, this is done to prevent approving unapproved comments during a reload: `@bors r+` approves the last commit in the PR, and during a reload that might be a commit pushed after the `r+`. The issue is prevented by pinning the SHA.

Reply:
Perfect. Thanks for the context. I'll update the comment.